### PR TITLE
Update docs for v8.10.0 release

### PR DESCRIPTION
**Pull request by JVerwolf**
Update the docs for v8.10.0,  Build Candidate 3

https://elasticsearch_99109.docs-preview.app.elstc.co/diff

### CODE REVIEW

**Comment by JVerwolf** [comment]
I'm unsure why this is failing, could it be due to [this](https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request+build-docs/70208/console)?
```
14:15:54 INFO:build_docs:asciidoctor: WARNING: migration/migrate_8_10.asciidoc: line 50: id assigned to block already in use: change_pre_configured_cached_analyzer_components_to_use_indexversion_instead_of_version
```

**Comment by jrodewig** [comment]
@JVerwolf Yes — That's causing the build to break. I think you'll need to revert that anchor to `change_pre_configured_cached_analyzer_components_to_use_indexversion_instead_of_version-highlight`.

If you auto-generated the changes to this file using the [changelog automation tool](https://docs.elastic.dev/elasticsearch-team/onboarding/elasticsearch-changelog-process), I'm not why this anchor would have changed. It may be something to raise with ES Delivery.

**Comment by abdonpijpelink** [comment]
>  I'm not why this anchor would have changed.

We manually changed this ID in an earlier iteration of these release notes (https://github.com/elastic/elasticsearch/pull/98737#discussion_r1302836849), because the release notes generator produced a duplicate ID (the regression we see on this PR).

File: docs/changelog/98204.yaml [hunk 1]
```diff
@@ -6,8 +6,8 @@ issues: []
 highlight:
   title: Enable parallel knn search across segments
   body: |-
-    Elasticsearch has until now performed search sequentially across the
+    Elasticsearch has, until now, performed search sequentially across the
     segments within each shard. This change makes knn queries faster on shards
-    that are made of more than one segment, by rewriting and collecting each
+    that are made of more than one segment by rewriting and collecting each
```
Comment (JVerwolf) [inline docs/changelog/98204.yaml:11]: FYI: I modified the phrasing in this `.yaml` file ([as per the instructions](https://docs.elastic.dev/elasticsearch-team/release/release-management#curate-the-notes)).

**Review by lcawl — APPROVED** [review]
LGTM

**Pull request merged**